src/EnergyPlus/WeatherManager.hh

```diff
@@ -774,95 +774,98 @@ struct WeatherManagerData : BaseGlobalStruct
-    std::string WaterMainsTempsScheduleName;                 // water mains tempeature schedule name
-    bool wthFCGroundTemps;
-
-    int TotRunPers;           // Total number of Run Periods (Weather data) to Setup
-    int TotRunDesPers;        // Total number of Run Design Periods (Weather data) to Setup
-    int TotReportPers;        // Total number of reporting periods
-    int TotThermalReportPers; // Total number of thermal reporting periods
-    int TotCO2ReportPers;     // Total number of CO2 reporting periods
-    int TotVisualReportPers;  // Total number of visual reporting periods
-
-    int NumSpecialDays;
-    Array1D_int SpecialDayTypes; // To hold holiday types given in input file NOLINT(cert-err58-cpp)
-    Array1D_int WeekDayTypes;    // To hold Week day types using specified first day NOLINT(cert-err58-cpp)
-    Array1D_int DSTIndex;        // To hold DST Index based on weather file or input NOLINT(cert-err58-cpp)
-
-    int NumDataPeriods;
-
-    int NumIntervalsPerHour;
-
-    bool UseDaylightSaving;         // True if user says to use Weather File specified DaylightSaving Period
-    bool UseSpecialDays;            // True if user says to use Weather File specified Special Days for current RunPeriod
-    bool UseRainValues;             // True if rain values from weather file are to be used
-    bool UseSnowValues;             // True if snow values from weather file are to be used
-    bool EPWDaylightSaving;         // True if a DaylightSaving Time Period is input (EPW files)
-    bool IDFDaylightSaving;         // True if a DaylightSaving Time Period is input (IDF files)
-    bool DaylightSavingIsActive;    // True if a DaylightSavingPeriod should be used for Environment
-    bool WFAllowsLeapYears;         // True if the Weather File (WF) header has "Yes" for Leap Years
-    int curSimDayForEndOfRunPeriod; // normal=number days in sim, but different when repeating runperiods or multi-year files
-    int Envrn;                      // Counter for environments
-    int NumOfEnvrn;                 // Number of environments to be simulated
-    int NumEPWTypExtSets;           // Number of Typical/Extreme on weather file.
-    int NumWPSkyTemperatures;       // Number of WeatherProperty:SkyTemperature items in input file
+    bool GetBranchInputOneTimeFlag = true;
+    bool GetEnvironmentFirstCall = true;
+    bool PrntEnvHeaders = true;
+    bool FirstCall = true;                 // Some things should only be done once
+    bool WaterMainsParameterReport = true; // should only be done once
+    bool PrintEnvrnStamp = false;          // Set to true when the environment header should be printed
+    bool PrintDDHeader = true;
+
+    int YearOfSim = 1;                     // The Present year of Simulation.
+    int NumDaysInYear = 365;               // TODO: removed const from this until leap year behavior is reviewed
```

Comment:
This was originally declared `const` which then triggered a custom check that this should be `constexpr` which meant moving `NumDaysInYear` out of `state` which meant changing every use of `NumDaysInYear` which meant `state` became an unused parameter in `getGroundTemp` functions which then cascaded into all sorts of nonsense. So leaving `NumDaysInYear` here and removing `const`. `NumDaysInYear` needs further investigation related to #8772, #8975, and #7240.

Reply:
In develop it is `const`, but custom_check is **not** failing in develop on it?  I guess it's because it's a slightly different looking phrase, where the value isn't actually assigned to the const on the same line.  But once it "looks like" `T const x = a`, then the custom_check starts failing.  

Either way, it's obvious what's going on here, and your choice to stop here is perfectly fine.  We can do another deeper dive on this variable in a dedicated PR where scope will creep further and further.